platform/entanglement/src/capabilities/LocationCapability.js

```diff
@@ -0,0 +1,77 @@
+/*global define */
+
+define(
+    function () {
+        "use strict";
+
+        function LocationCapability(domainObject) {
+            this.domainObject = domainObject;
+            return this;
+        }
+
+        /**
+         * Persist the current location of the current domain object as it's
+         * primary location.  Returns a promise.
+         */
+        LocationCapability.prototype.persistLocation = function () {
```

Comment:
I've tweaked the naming/wording of this for clarity (push coming shortly) the reason I choose this over an inline `locate` function is because I prefer the capability to control the behavior for both retrieving location and setting location.  It also simplifies testing of code that depends on the locationcapability because that code doesn't have to worry about the underlying implementation.